be/src/exec/pipeline/scan/connector_scan_operator.cpp

```diff
@@ -27,42 +27,47 @@ namespace starrocks::pipeline {
 // ==================== ConnectorScanOperatorFactory ====================
 
 struct ConnectorScanOperatorIOTasksMemLimiter {
-    mutable std::shared_mutex lock;
-
-    size_t dop = 0;
-    int64_t mem_limit = 0;
-    int64_t running_chunk_source_count = 0;
-    int64_t estimated_mem_usage_update_count = 0;
-    double estimated_mem_usage_per_chunk_source = 0;
-
-    int available_chunk_source_count() const {
-        std::shared_lock<std::shared_mutex> L(lock);
-
-        int64_t max_count = mem_limit / estimated_mem_usage_per_chunk_source;
-        int64_t avail_count = (max_count - running_chunk_source_count) / static_cast<int64_t>(dop);
-        avail_count = std::max<int64_t>(avail_count, 0);
-        if (avail_count == 0 && running_chunk_source_count == 0) {
-            avail_count = 1;
+    mutable std::mutex lock;
+
+    const int64_t dop = 0;
+    std::atomic<int64_t> scan_mem_limit = 0;
+    std::atomic<int64_t> running_chunk_source_count = 0;
+    std::atomic<int64_t> chunk_source_mem_bytes = 0;
+    int64_t chunk_source_mem_bytes_update_count = 0;
```

Comment:
It seems `lock` now only guards this variable. Why not make this an atomic and remove 
 `lock`?

Reply:
because to computer avarage, you have to update two variables in a atomic way.